src/transactions/InvokeHostFunctionOpFrame.cpp

```diff
@@ -93,35 +245,48 @@ InvokeHostFunctionOpFrame::doApply(AbstractLedgerTxn& ltx, Config const& cfg)
     {
         // Load without record for readOnly to avoid writing them later
         auto ltxe = ltx.loadWithoutRecord(lk);
+        size_t nByte{0};
         if (ltxe)
         {
-            ledgerEntryCxxBufs.emplace_back(toCxxBuf(ltxe.current()));
+            auto buf = toCxxBuf(ltxe.current());
+            nByte = buf.data->size();
+            ledgerEntryCxxBufs.emplace_back(std::move(buf));
         }
+        metrics.noteReadEntry(lk, nByte);
     }
     for (auto const& lk : mInvokeHostFunction.footprint.readWrite)
     {
         auto ltxe = ltx.load(lk);
+        size_t nByte{0};
         if (ltxe)
         {
-            ledgerEntryCxxBufs.emplace_back(toCxxBuf(ltxe.current()));
+            auto buf = toCxxBuf(ltxe.current());
+            nByte = buf.data->size();
+            ledgerEntryCxxBufs.emplace_back(std::move(buf));
         }
+        metrics.noteWriteEntry(lk, nByte);
     }
 
     InvokeHostFunctionOutput out;
     try
     {
+        auto timeScope = metrics.getExecTimer();
         out = rust_bridge::invoke_host_function(
             toCxxBuf(mInvokeHostFunction.function),
             toCxxBuf(mInvokeHostFunction.parameters),
             toCxxBuf(mInvokeHostFunction.footprint), toCxxBuf(getSourceID()),
             getLedgerInfo(ltx, cfg), ledgerEntryCxxBufs);
+        metrics.mSuccess = true;
     }
     catch (std::exception&)
     {
         innerResult().code(INVOKE_HOST_FUNCTION_TRAPPED);
         return false;
     }
 
+    metrics.mCpuInsn = out.cpu_insns;
```

Comment:
we are not doing this accounting if the invoke traps which may hide some patterns. Any way we can still get a read on this in all cases? (I think the limitation is coming from relying on the result for this)

Reply:
Yeah, we get no output in case of an exception, so there's nothing to read this out of. The information channel in Rust errors translated to C++ exceptions is quite narrow, I don't think it'd be feasible to convey that way.

Reply:
ok. I guess we'll still have the overall timing so it's not that bad. Still annoying that we can't know how much actual instructions we executed for an entire ledger